js/services/engine.ts

```diff
@@ -57,7 +57,9 @@ function processObjectDataRequests(components, flowKey) {
 
                 if (component.fileDataRequest) {
 
-                    return fileDataRequest(component.id, component.fileDataRequest, flowKey, limit);
+                    if (component.componentType !== 'file-upload') {
```

Comment:
This could just be one statement - if (component.fileDataRequest && component.componentType !== 'file-upload')

Reply:
no it can't as it will enter in to the else if it's a file-upload and that breaks javascript and stop things working as:
`Uncaught TypeError: Cannot read property 'listFilter' of null
    at Object.n.dispatchDataRequest (ajax.ts:34)
    at Object.n.dispatchObjectDataRequest (ajax.ts:185)
    at Object.n.objectDataRequest (engine.ts:1172)
    at engine.ts:67
    at Array.map (<anonymous>)
    at x (engine.ts:47)
    at L (engine.ts:198)
    at Object.<anonymous> (engine.ts:529)
    at j (jquery-2.1.4.min.js:2)
    at Object.fireWith [as resolveWith] (jquery-2.1.4.min.js:2)`